src/common/low_precision_transformations/src/strided_slice.cpp

```diff
@@ -18,62 +18,81 @@ namespace low_precision {
 namespace {
 
 std::shared_ptr<opset1::Constant> stridedSliceDeqConstant(
-    const std::shared_ptr<ngraph::Node> strSlice,
-    const std::shared_ptr<ngraph::Node> dequantizaitonConstant) {
-    auto constant = ov::as_type_ptr<ngraph::opset1::Constant>(dequantizaitonConstant);
-    auto constantShape = constant->get_shape();
-    if (shape_size(constantShape) == 1ul) {
+    const std::shared_ptr<Node> node,
+    const std::shared_ptr<Node> dequantizaiton_constant) {
+    const auto constant = ov::as_type_ptr<ov::op::v0::Constant>(dequantizaiton_constant);
+    const auto& original_constant_shape = constant->get_shape();
+    if (shape_size(original_constant_shape) == 1ul) {
         return NetworkHelper::toScalar(constant);
     }
 
-    const auto stridedSlicePShape = strSlice->get_input_partial_shape(0);
-    const size_t rank = stridedSlicePShape.rank().get_length();
-    if (rank != constantShape.size()) {
-        ngraph::Shape newConstantShape;
-        if (ngraph::shape_size(constantShape) == 1) {
-            newConstantShape = ngraph::Shape(rank, 1);
-        } else {
-            newConstantShape = constantShape;
-
-            // case when constShape without batch
-            if ((constantShape.size() > 1) &&
-                (constantShape.size() < rank)) {
-                newConstantShape.insert(newConstantShape.begin(), 1);
-            }
+    // step #1: align shapes
+    std::shared_ptr<ov::op::v0::Constant> new_constant = constant;
+    const size_t rank = node->get_input_partial_shape(0).size();
+    Shape new_constant_shape = original_constant_shape;
+    if (rank != new_constant_shape.size()) {
+        // case when constant shape without batch
+        if (original_constant_shape.size() < rank) {
+            new_constant_shape.insert(new_constant_shape.begin(), 1);
         }
```

Comment:
Let's support all possible cases.
```suggestion
        while (new_constant_shape.size() < rank) {
            new_constant_shape.insert(new_constant_shape.begin(), 1);
        }
```

Reply:
We can not add all dimensions at the begining. Branch is added in the begining, other dimensions are added during broadcast.